ddl/job_table.go

```diff
@@ -309,6 +314,121 @@ func (d *ddl) markJobProcessing(sess *session, job *model.Job) error {
 	return errors.Trace(err)
 }
 
+func (d *ddl) startDispatchBackfillJobsLoop() {
+	d.backfillCtx.jobCtxMap = make(map[int64]*JobContext)
+	d.backfillCtx.backfillCtxMap = make(map[int64]struct{})
+
+	var notifyBackfillJobByEtcdCh clientv3.WatchChan
+	if d.etcdCli != nil {
+		notifyBackfillJobByEtcdCh = d.etcdCli.Watch(d.ctx, addingBackfillJob)
+	}
+	ticker := time.NewTicker(500 * time.Millisecond)
+	defer ticker.Stop()
+	for {
+		if isChanClosed(d.ctx.Done()) {
+			return
+		}
+		select {
+		case <-d.backfillJobCh:
+		case <-ticker.C:
+		case _, ok := <-notifyBackfillJobByEtcdCh:
+			if !ok {
+				logutil.BgLogger().Warn("[ddl] start backfill worker watch channel closed", zap.String("watch key", addingBackfillJob))
+				notifyBackfillJobByEtcdCh = d.etcdCli.Watch(d.ctx, addingBackfillJob)
+				time.Sleep(time.Second)
+				continue
+			}
+		case <-d.ctx.Done():
+			return
+		}
+		d.loadBackfillJobAndRun()
+	}
+}
+
+func (d *ddl) getTableByTxn(store kv.Storage, schemaID, tableID int64) (*model.DBInfo, table.Table, error) {
+	var tbl table.Table
+	var dbInfo *model.DBInfo
+	err := kv.RunInNewTxn(d.ctx, store, false, func(ctx context.Context, txn kv.Transaction) error {
+		t := meta.NewMeta(txn)
+		var err1 error
+		dbInfo, err1 = t.GetDatabase(schemaID)
+		if err1 != nil {
+			return errors.Trace(err1)
+		}
+		tblInfo, err1 := getTableInfo(t, tableID, schemaID)
+		if err1 != nil {
+			return errors.Trace(err1)
+		}
+		tbl, err1 = getTable(store, schemaID, tblInfo)
+		return errors.Trace(err1)
+	})
+	return dbInfo, tbl, err
+}
+
+func (d *ddl) loadBackfillJobAndRun() {
+	isDistReorg := variable.DDLEnableDistributeReorg.Load()
+	if !isDistReorg {
+		return
+	}
+	se, err := d.sessPool.get()
+	defer d.sessPool.put(se)
+	if err != nil {
+		logutil.BgLogger().Fatal("dispatch backfill jobs loop get session failed, it should not happen, please try restart TiDB", zap.Error(err))
+	}
+	sess := newSession(se)
+
+	runningJobIDs := d.backfillCtxJobIDs()
+	if len(runningJobIDs) >= reorgWorkerCnt {
+		d.backfillCtx.Unlock()
+		return
+	}
+
+	// TODO: Add ele info to distinguish backfill jobs.
+	bJobs, err := GetBackfillJobsForOneEle(sess, 1, runningJobIDs, InstanceLease)
```

Comment:
The `batch` size always equals to 1 for `GetBackfillJobsForOneEle` func. Do we have further plan for this param?